day2.py

```diff
@@ -0,0 +1,109 @@
+import pandas as pd
+import numpy as np
+
+
+### TASK 1 ###
+
+# import the data that I saved as a csv
+path = 'D:/'
+day2 = pd.read_csv(path + 'day2_input.csv', header = 0)
+
+# slit the game games into two cols, one result that the opponant plays, one that I play
+day2[['opponant', 'you']] = day2['result'].str.split(' ', 1, expand=True)
+
+# create columns that hold the amount of points the player got for playing that option
+day2['opp_score'] = np.where(
+    day2['opponant'] == 'A', 1, np.where(
+        day2['opponant'] == 'B', 2, np.where(
+            day2['opponant'] == 'C', 3, 0)))
+
+day2['my_score'] = np.where(
+    day2['you'] == 'X', 1, np.where(
+        day2['you'] == 'Y', 2, np.where(
+            day2['you'] == 'Z', 3, 0)))
+
+# create column and populate with who wins the game based on the rules of the game
+day2['outcome'] = np.where((
+    ((day2['opp_score'] == 1) & (day2['my_score'] == 2)) |
+    ((day2['opp_score'] == 2) & (day2['my_score'] == 3)) | 
+    ((day2['opp_score'] == 3) & (day2['my_score'] == 1))),
+    'i win',
+        np.where(
+            ((day2['opp_score'] == 1) & (day2['my_score'] == 1)) | 
+            ((day2['opp_score'] == 2) & (day2['my_score'] == 2)) | 
+            ((day2['opp_score'] == 3) & (day2['my_score'] == 3)),
+        'draw',         
+        np.where(
+            ((day2['opp_score'] == 1) & (day2['my_score'] == 3)) | 
+            ((day2['opp_score'] == 2) & (day2['my_score'] == 1)) | 
+            ((day2['opp_score'] == 3) & (day2['my_score'] == 2)),
+        'opp wins', 0))
+)
+ 
+# creating a column and allocating points to players based on the result of the game (6 for win, 3 or draw, 0 for loss)
+day2['opp_outcome'] = np.where(
+    day2['outcome'] == 'opp wins', 6, np.where(
+        day2['outcome'] == 'draw', 3, np.where(
+            day2['outcome'] == 'i win', 0, 0)))
+
+day2['my_outcome'] = np.where(
+    day2['outcome'] == 'opp wins', 0, np.where(
+        day2['outcome'] == 'draw', 3, np.where(
+            day2['outcome'] == 'i win', 6, 0)))
```

Comment:
Could you create a function that does all of this at once? 
e.g. This could use dictionaries again that define how combinations result in outcome scores. Something like: 
- replacing `i win`, `draw` and `opp wins` for the combinations with integers;
- takes two columns of scores as inputs;
- outputs a new column to the dataframe;

```
def winner_scores(df, player1_col, player2_col, new_col_name):
    # Define combinations and their outcomes for player1 and player2
   df[new_col_name] = # applies combination based on input column scores
    return df 
```